frontend/models/product.ts

```diff
@@ -90,10 +100,46 @@ function getVariants(shopifyProduct: NonNullable<FindProductQuery['product']>) {
          specifications: variant.specifications?.value ?? null,
          warranty: variant.warranty?.value ?? null,
          crossSellVariants: getCrossSellVariants(variant),
+         disableWhenOOS: variant.disableWhenOOS?.value === 'true',
       };
    });
 }
 
+function getImages(
+   shopifyImages: NonNullable<FindProductQuery['product']>['images'],
+   variants: ReturnType<typeof getVariants>
+) {
+   return shopifyImages.nodes.filter(
+      (image) =>
+         !variants.find(
+            (variant) =>
+               image.altText &&
+               variant.sku === image.altText &&
+               variant.disableWhenOOS &&
+               variant.quantityAvailable != null &&
+               variant.quantityAvailable <= 0
+         )
+   );
+}
+
+function getOptions(
+   shopifyOptions: NonNullable<FindProductQuery['product']>['options'],
+   activeVariants: ReturnType<typeof getVariants>
+) {
+   return shopifyOptions.map((option) => ({
+      ...option,
+      values: option.values.filter((value) =>
```

Comment:
What happens if `values` ends up being an empty array? Looking at `ProductOption.tsx`, it seems like we'd get some strange UI, or array indexing errors. Maybe there's a conditional that I'm missing somewhere that guards against this.